src/metaopt/io/database/mongodb.py

```diff
@@ -50,6 +50,7 @@ def initiate_connection(self):
         try:
             conn = MongoClient(self.uri)
             self.conn = conn
+            self.conn.admin.command('ismaster')
```

Comment:
That's a good example of why detailed commit messages are necessary. I believe that would be a good time to read and apply what I suggested [in this comment](https://github.com/mila-udem/metaopt/pull/8#issuecomment-344464658). In this case, the commit message should explain why we need to set ismaster and maybe also the alternatives and why command('ismaster') is better. 

Writing detailed commit messages does slow down the process of committing code but it significantly speeds up review and maintenance. Also, you will sometimes realize while writing the commit message that something is wrong or missing in the commit. Many of your commits in this PR have been undo/redo, which is a sign that what you were doing was not totally clear to you. It was my fault many times, because my comments were not detailed enough, but if you took the time to write down the commit message you could have realized that informations was missing and you would have asked me for clarifications before committing the changes.